### ISSUE TITLE
Bitswap is getting stuck in tests

### ISSUE THREAD

**Issue by jbenet** [ipfs/go-ipfs#1249]
This is definitely bitswap getting stuck in a test: https://travis-ci.org/ipfs/go-ipfs/jobs/63095960

Hopefully #1218 fixes it.

cc @whyrusleeping

**Comment by jbenet** [comment]
This is happening _a lot_ now. I re-ran the same build 10 times and it halted the same way every time.

**Comment by jbenet** [comment]
- https://travis-ci.org/ipfs/go-ipfs/jobs/63095953
- https://travis-ci.org/ipfs/go-ipfs/jobs/63095955
- https://travis-ci.org/ipfs/go-ipfs/jobs/63095547
- https://travis-ci.org/ipfs/go-ipfs/jobs/63095960
- https://travis-ci.org/ipfs/go-ipfs/jobs/63095959

**Comment by whyrusleeping** [comment]
tests are a good deal more stable in #1218 one of the things i did was make the bitswap tests not happen in parallel, there were still lingering goroutines that would pile up and exponentially degrade performance in the larger tests, for example, we spawn a goroutine per message sent in the test net so the 'deliver' function can wait before handing the message off, these messages have to be handled one at a time, and the slower the code runs, the more of them build up, causing things to be even slower, and more goroutines to build up. I should probably fix up the testnet a bit, but that seemed out of scope for #1218

**Comment by whyrusleeping** [comment]
i think this has been pretty much fixed entirely. I no longer see it fail
